src/ngraph/runtime/cpu/op/quantize.cpp

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ngraph/runtime/cpu/op/quantize.hpp"
+
+ngraph::op::Quantize::Quantize(std::shared_ptr<Node> input,
+                               std::shared_ptr<Node> min,
+                               std::shared_ptr<Node> max,
+                               const element::Type& type)
+    : RequiresTensorViewArgs("Quantize", {input, min, max})
+    , m_element_type(type)
+{
```

Comment:
If this goes into the core opset we'll need to add a check for the shapes/ETs of `min` and `max`. We'll also need to decide what element types we want to allow, both for the unquantized input (TF just allows float32) and the quantized output (TF has several allowed types).

For experimental purposes in the meantime, I'd be okay with targeting this directly (keeping things under `runtime/cpu`), and putting some safety logic in the bridge.

Reply:
Yes agreed, the ops dont have proper checks for ETs/shapes,  because my understanding was to have some temporary cpu specific ops until core has these ops, just so that i can break dependency from core and test my code. I am not sure what the plan is now. Maybe @jbobba can answer better if we want the bridge to call into cpu ops directly or the core can use these ops as a prototype and improve on it.